container/id/id.go

```diff
@@ -1,90 +1,112 @@
 package cid
 
 import (
-	"bytes"
 	"crypto/sha256"
-	"errors"
+	"fmt"
 
 	"github.com/mr-tron/base58"
 	"github.com/nspcc-dev/neofs-api-go/v2/refs"
 )
 
-// ID represents v2-compatible container identifier.
-type ID refs.ContainerID
-
-// NewFromV2 wraps v2 ContainerID message to ID.
+// ID represents NeoFS container identifier.
 //
-// Nil refs.ContainerID converts to nil.
-func NewFromV2(idV2 *refs.ContainerID) *ID {
-	return (*ID)(idV2)
-}
-
-// New creates and initializes blank ID.
+// ID is mutually compatible with github.com/nspcc-dev/neofs-api-go/v2/refs.ContainerID
+// message. See ReadFromV2 / WriteToV2 methods.
 //
-// Defaults:
-//  - value: nil.
-func New() *ID {
-	return NewFromV2(new(refs.ContainerID))
-}
+// Instances can be created using built-in var declaration.
+//
+// Note that direct typecast is not safe and may result in loss of compatibility:
+// 	_ = ID([32]byte) // not recommended
+type ID [sha256.Size]byte
 
-// SetSHA256 sets container identifier value to SHA256 checksum of container body.
-func (id *ID) SetSHA256(v [sha256.Size]byte) {
-	(*refs.ContainerID)(id).SetValue(v[:])
+// ReadFromV2 reads ID from the refs.ContainerID message.
+//
+// See also WriteToV2.
+func (id *ID) ReadFromV2(m refs.ContainerID) error {
+	return id.Decode(m.GetValue())
 }
 
-// ToV2 returns the v2 container ID message.
+// WriteToV2 writes ID to the refs.ContainerID message.
+// The message must not be nil.
 //
-// Nil ID converts to nil.
-func (id *ID) ToV2() *refs.ContainerID {
-	return (*refs.ContainerID)(id)
+// See also ReadFromV2.
+func (id ID) WriteToV2(m *refs.ContainerID) {
+	m.SetValue(id[:])
 }
 
-// Equal returns true if identifiers are identical.
-func (id *ID) Equal(id2 *ID) bool {
-	return bytes.Equal(
-		(*refs.ContainerID)(id).GetValue(),
-		(*refs.ContainerID)(id2).GetValue(),
-	)
+// Encode encodes ID into 32 bytes of dst.
+//
+// Zero ID is all zeros.
+//
+// Encode panics if dst length is less than 32.
+//
+// See also Decode.
+func (id ID) Encode(dst []byte) {
+	copy(dst, id[:])
 }
 
-// Parse parses string representation of ID.
+// Decode decodes src bytes into ID.
 //
-// Returns error if s is not a base58 encoded
-// ID data.
-func (id *ID) Parse(s string) error {
-	data, err := base58.Decode(s)
-	if err != nil {
-		return err
-	} else if len(data) != sha256.Size {
-		return errors.New("incorrect format of the string container ID")
+// Decode expects that src has 32 bytes length. If the input is malformed,
+// Decode returns an error. In this case src remains unchanged.
```

Comment:
```suggestion
// Decode returns an error. In this case ID remains unchanged.
```
Statement that `src remains unchanged` is also needed.